src/client/httpClient.ts

```
@@ -123,34 +123,42 @@ export class HTTPClient {
 
     this.baseApiPath = options.baseApiPath ?? '';
     this.applicationToken = options.applicationToken;
-    this.authHeaderName = options.authHeaderName || DEFAULT_AUTH_HEADER;
+    this.authHeaderName = options.authHeaderName || process.env.AUTH_HEADER_NAME || DEFAULT_AUTH_HEADER;
     this.logLevel = options.logLevel || DEFAULT_LOG_LEVEL;
   }
 
-  async _request(options: AxiosRequestConfig) {
+  async _request(requestInfo: AxiosRequestConfig) {
     try {
+      //console.log("URL : " + requestInfo.url);
+      //console.log("Method : " + requestInfo.method);
+      //console.log("data : " + JSON.stringify(requestInfo.data));
       const response = await axiosAgent({
-        url: options.url,
-        data: options.data,
-        params: options.params,
-        method: options.method || DEFAULT_METHOD,
-        timeout: options.timeout || DEFAULT_TIMEOUT,
+        url: requestInfo.url,
+        data: requestInfo.data,
+        params: requestInfo.params,
+        method: requestInfo.method || DEFAULT_METHOD,
+        timeout: requestInfo.timeout || DEFAULT_TIMEOUT,
         headers: {
           [this.authHeaderName]: this.applicationToken
         }
       });
-      if (response.data.data) {
-        return {
-          status: response.status,
-          ...response.data
-        };
-      }
+      //const resp = {
       return {
-        status: response.status,
-        data: response.data
+        status: response.data.status,
+        data: response.data.data,
+        errors: response.data.errors
       };
+      //console.log(JSON.stringify(resp));
+      //return resp;
     } catch (e: any) {
-      throw new AstraError(e.response?.data?.description || e?.message, e?.response);
+      console.log(e);
+       return {
+        errors: [
+          {
+            message: "Server call failed, please retry!"
```

The error message is swallowed, lets set message to be `e.message` and also add `error:e` to the object?

---

I set the message to be e.message. 

But I'm not clear on this part of the message. Could you please elaborate a little bit more on this?

> 'also add error:e to the object?'